pkg/audit/audit.go

```diff
@@ -16,127 +16,225 @@ package audit
 
 import (
 	"context"
+	"fmt"
+	"net/http"
 	"sync"
+	"time"
 
-	"github.com/kubesphere/kubeeye/apis/kubeeye/v1alpha1"
+	"k8s.io/client-go/util/workqueue"
+
+	kubeeyev1alpha2 "github.com/kubesphere/kubeeye/apis/kubeeye/v1alpha2"
+	"github.com/kubesphere/kubeeye/constant"
 	"github.com/kubesphere/kubeeye/pkg/kube"
-	"github.com/kubesphere/kubeeye/pkg/regorules"
-	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/json"
 	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-var (
-	workloads = "data.kubeeye_workloads_rego"
-	rbac      = "data.kubeeye_RBAC_rego"
-	nodes     = "data.kubeeye_nodes_rego"
-	events    = "data.kubeeye_events_rego"
-	certexp   = "data.kubeeye_certexpiration"
-)
+type Audit struct {
+	TaskQueue   workqueue.RateLimitingInterface
```

Comment:
you should call TaskQueue.Forget() or TaskQueue.AddRateLimited() after you processed an item